packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -809,7 +853,166 @@ class _CheckboxPainter extends ToggleablePainter {
         }
       }
     }
+    switch (_designSpec!) {
+      case _DesignSpec.cupertino:
+        // Apply effect to darken checkbox when pressed.
+        if (!reaction.isDismissed){
+          final Paint paint = Paint()
+            ..color = Colors.transparent.withOpacity(0.05);
+          final Rect outer = _outerRectAt(origin, 1.0);
+          _drawBox(canvas, outer, paint, activeSide);
+        }
+        if (isFocused) {
+          final Rect focusOuter = _outerRectAt(origin, 1.0).inflate(1);
+          final Paint borderPaint = Paint()
+            ..color = focusColor
+            ..style = PaintingStyle.stroke
+            ..strokeWidth = 3.5;
+          _drawBox(canvas, focusOuter, borderPaint, activeSide);
+        }
+      case _DesignSpec.material:
+        break;
+    }
+  }
+}
+
+// Hand coded defaults for iOS/macOS checkbox
+// TODO(victorsanni): Apply theme for widget state where app is open but inactive.
```

Comment:
Are you still planning to do this in this PR?

Reply:
I am not sure. This will probably require further discussion on if I should do this if/when WidgetState gets updated or bypass that by somehow adding an `AppLifeCycleStateListener` to know when the app is open but inactive.